lib/datatables_crud/datatables/active_record_datatable.rb

```diff
@@ -30,27 +30,49 @@ def sort_options
       end
 
       def count_options
-        (@options || {}).reject { |k, v| %w(limit offset).include? k.to_s }
+        (@options || {}).reject { |k, v| %w(limit offset order).include? k.to_s }
       end
 
       def count
         if params[:search].try(:[], :value).present? and search_columns.present?
-          @count ||= prepared_clazz.where(search_columns.map { |v| "#{v} like :search" }.join(' OR '), search: "%#{params[:search][:value]}%").count(count_options)
+          @count ||= count_of( prepared_clazz.where(search_columns.map { |v| "#{v} like :search" }.join(' OR '), search: "%#{params[:search][:value]}%") )
         else
           total_count
         end
       end
 
       def total_count
-        @total_count ||= prepared_clazz.count(count_options)
+        @total_count ||= count_of(prepared_clazz)
+      end
+
+      def count_of(objects)
+        objects.where(count_options[:conditions]).count
       end
 
       def records
+        @records_exist = @records.present?
         if params[:search].try(:[], :value).present? and search_columns.present?
-          @records ||= prepared_clazz.where(search_columns.map { |v| "#{v} like :search" }.join(' OR '), search: "%#{params[:search][:value]}%").all(@options)
+          @records ||= prepared_clazz.where(search_columns.map { |v| "#{v} like :search" }.join(' OR '), search: "%#{params[:search][:value]}%")
         else
-          @records ||= prepared_clazz.all(@options)
+          @records ||= prepared_clazz.all
         end
+
+        apply_options
+
+        @records
+      end
+
+      def apply_options
+        return if @records_exist
+        return unless @records
+
+        if @options[:conditions].present?
+          @records = @records.where(@options[:conditions])
+        end
+
+        @options.reject {|k, v| k.to_s == "conditions" }.each do |k, v|
```

Comment:
I prefer this approach: https://github.com/anikolskiy/datatables_crud/blob/bootstrap_3_simple_form_rails_5/lib/datatables_crud/datatables/active_record_datatable.rb#L50

It's way more easy to understand